src/pyrqlite/cursors.py

```diff
@@ -277,8 +277,12 @@ def fetchone(self):
 
     def fetchmany(self, size=None):
         if size is None:
-            size = self.arraysize
-        raise NotImplementedError(self)
+            size = max(self.arraysize, len(self._rows))
+        size = min(len(self._rows), size)
```

Comment:
The size calculation looks like in could be incorrect for some corner cases. Also, the [fetchmany documentation](https://www.python.org/dev/peps/pep-0249/#fetchmany) says to use arraysize when size has not been specified, so I'd like to conform to that. How about something like this:
```python
    def fetchmany(self, size=None):
        remaining = self.arraysize if size is None else size
        remaining = min(remaining, self.rowcount - self.rownumber)
        for i in range(remaining):
            rows.append(self.fetchone())
        return rows
```

Reply:
I've merged the Row slice change in aa562e4942531e96a13b927b69be4bdf9ed6cc43, and the cursor factory change in 9da1eb12c66e73b425b70523984517ec744d3a5a.